src/Nethermind/Nethermind.Trie/VisitingOptions.cs

```diff
@@ -21,5 +21,10 @@ public class VisitingOptions
         /// Maximum number of threads that will be used to visit the trie.
         /// </summary>
         public int MaxDegreeOfParallelism { get; init; } = 1;
+
+        /// <summary>
+        /// Specify that this is a full table scan and should optimize for it.
+        /// </summary>
+        public bool FullDbScan { get; set; }
```

Comment:
maybe this should be `true` by default? Most of trie visits are doing full trie scan.

Reply:
Actually, this flag probably should be in `ITreeVisitor` instead.